hq/app/logic/Retry.scala

```diff
@@ -3,22 +3,29 @@ package logic
 import utils.attempt.{Attempt, Failure}
 
 import scala.concurrent.ExecutionContext
+import scala.concurrent.duration.FiniteDuration
 object Retry {
 
-  def until[A](body: => Attempt[A], predicate: A => Boolean, failureMessage: String, maxAttempt: Int = 10)(implicit ec: ExecutionContext): Attempt[A] = {
+  /**
+    * execute the body until the predicate is satisfied or maxAttempt is reached, with the delay between each body execution.
+    */
+  def until[A](body: => Attempt[A], predicate: A => Boolean, failureMessage: String, delay: FiniteDuration , maxAttempt: Int = 10)(implicit ec: ExecutionContext): Attempt[A] = {
+    def determineBody(tryCnt: Int) : Attempt[A] =  if (tryCnt == 0) body else body.delay(delay)
```

Comment:
Is there any reason not to inline this? The function definition is larger than the logic and it's separated so it can be tested. This means the indirection doesn't seem to achieve much?

Reply:
👍